networkit/cpp/centrality/DynApproxBetweenness.cpp

```diff
@@ -7,172 +8,316 @@
 #include <networkit/distance/Diameter.hpp>
-#include <networkit/graph/GraphTools.hpp>
-#include <networkit/distance/DynDijkstra.hpp>
 #include <networkit/distance/DynBFS.hpp>
-#include <networkit/auxiliary/Log.hpp>
-#include <networkit/auxiliary/NumericTools.hpp>
-
+#include <networkit/distance/DynDijkstra.hpp>
+#include <networkit/graph/GraphTools.hpp>
 
+// networkit-format
 namespace NetworKit {
 
-DynApproxBetweenness::DynApproxBetweenness(const Graph& G, const double epsilon, const double delta, const bool storePredecessors, const double universalConstant) : Centrality(G, true),
-storePreds(storePredecessors), epsilon(epsilon), delta(delta), universalConstant(universalConstant) {
+DynApproxBetweenness::DynApproxBetweenness(const Graph &G, double epsilon, double delta,
+                                                   bool storePredecessors, double universalConstant)
+    : Centrality(G, true), epsilon(epsilon), delta(delta), storePreds(storePredecessors),
+      universalConstant(universalConstant) {}
+
+inline count DynApproxBetweenness::DFS(node u, count j, Graph &sccDAG,
+                                           std::vector<bool> &marked, std::vector<node> &sorted) {
+    marked[u] = true;
+    sccDAG.forNeighborsOf(u, [&](node v) {
+        if (!marked[v]) {
+            marked[v] = true;
+            j = DFS(v, j, sccDAG, marked, sorted);
+        }
+    });
+    j--;
+    sorted[j] = u;
+    return j;
 }
 
+count DynApproxBetweenness::computeVDdirected() {
+    clock_t begin = clock();
+    StronglyConnectedComponents scc(G);
+    scc.run();
+    clock_t end = clock();
+    INFO("Time to run dyn conn comp: ", end - begin);
+    begin = clock();
+    count sccNum = scc.numberOfComponents();
+    INFO("Number of nodes: ", G.upperNodeIdBound(), ", number of components: ", sccNum);
+    Graph sccDAG(sccNum, true, true);
+    G.forEdges([&](node u, node v) {
+        if (scc.componentOfNode(u) != scc.componentOfNode(v)) {
+            if (!sccDAG.hasEdge(scc.componentOfNode(u) - 1, scc.componentOfNode(v) - 1)) {
+                sccDAG.addEdge(scc.componentOfNode(u) - 1,
+                               scc.componentOfNode(v)
+                                   - 1); // maybe sccSizes[scc.componentOfNode(v)] ?
+            }
+        }
+    });
+    end = clock();
+    INFO("Time to create scc DAG: ", end - begin);
+    begin = clock();
+    count j = sccNum;
+    std::vector<node> sorted(sccNum);
+    std::vector<bool> marked(sccNum);
+
+    sccDAG.forNodes([&](node c) {
+        if (!marked[c] && sccDAG.degreeIn(c) == 0) {
+            marked[c] = true;
+            j = DFS(c, j, sccDAG, marked, sorted);
+        }
+    });
+
+    end = clock();
+    INFO("Time for topological sorting: ", end - begin);
+    begin = clock();
+    std::vector<bool> marked1(sccNum);
+    marked1.resize(G.upperNodeIdBound(), false);
+    std::vector<bool> marked2(sccNum);
+    marked2.resize(G.upperNodeIdBound(), false);
```

Comment:
Why not `std::vector<bool> marked1(G.upperNodeIdBound()), marked2(G.upperNodeIdBound());` ?